### PR TITLE
fix: Converting configuration API response from an array to dictionary

### PR DESCRIPTION
**Pull request by beiwei30**
Signed-off-by: Ian Luo <ian.luo@gmail.com>

# Description

fix issue #4556

## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by beiwei30** [comment]
Have to wait for dapr/components-contrib#1771 merged. Pls. don't merge for now.

@LaurenceLiZhixin pls. take a look.

**Comment by yaron2** [comment]
@beiwei30 see build errors:

```
Error: pkg/grpc/api.go:1554:15: cannot use k (variable of type int) as type string in map index
Error: pkg/grpc/api.go:1577:9: cannot use k (variable of type int) as type string in map index
```

**Comment by LaurenceLiZhixin** [comment]
Thanks for @beiwei30 , the modification seems to me to be in the right direction. Please update conpoment version after https://github.com/dapr/components-contrib/pull/1771 is merged.

@yaron2 The build errors occurs because the component not updated, which is in line with expectations :)

**Comment by yaron2** [comment]
> 

The build is failing now due to another issue which is unrelated. @ItalyPaleAle will submit a PR to fix.

**Comment by yaron2** [comment]
@beiwei30 

```
Error: pkg/grpc/api.go:1554:15: cannot use k (variable of type int) as type string in map index
Error: pkg/grpc/api.go:1577:9: cannot use k (variable of type int) as type string in map index
```

**Comment by beiwei30** [comment]
> @beiwei30
> 
> ```
> Error: pkg/grpc/api.go:1554:15: cannot use k (variable of type int) as type string in map index
> Error: pkg/grpc/api.go:1577:9: cannot use k (variable of type int) as type string in map index
> ```

@yaron2 This is expected. Because the change made on component-contrib side (https://github.com/dapr/components-contrib/pull/1771) has not been merged yet. Once it's done, I need to bump up component-contrib version in order to pass compilation failure

**Comment by yaron2** [comment]
> > @beiwei30
> > ```
> > Error: pkg/grpc/api.go:1554:15: cannot use k (variable of type int) as type string in map index
> > Error: pkg/grpc/api.go:1577:9: cannot use k (variable of type int) as type string in map index
> > ```
> 
> @yaron2 This is expected. Because the change made on component-contrib side ([dapr/components-contrib#1771](https://github.com/dapr/components-contrib/pull/1771)) has not been merged yet. Once it's done, I need to bump up component-contrib version in order to pass compilation failure.
> 
> Right now I submit the following change to make build pass:
> 
> ```
> replace github.com/dapr/components-contrib => github.com/beiwei30/components-contrib v0.4.1-0.20220603163116-fa6cff30d029
> ```

@LaurenceLiZhixin @beiwei30 you're right, apologies :)

We will merge this change right after the code freeze (6/21) for 1.9 as we need more time to update the SDKs to match the new data structure.

**Comment by addjuarez** [comment]
ping @beiwei30

File: go.mod
Comment (berndverst) [inline go.mod:354]: Remove this please. We can make coordinate whatever change is required in the contrib repo first.

**Review by berndverst — CHANGES_REQUESTED** [review]
Looks good to me, but this depends on the change in dapr/contrib being merged first, and then the version of contrib needs to be imported here I think.

**Pull request closed without merging**
